### ISSUE TITLE
Integrade dynamic swallow please?

### ISSUE THREAD

**Issue by der-zonk** [bakkeby/dwm-flexipatch#292]
https://github.com/slavistan/dwm-dynamicswallow-patch

Use-case for me is:
Rxvt-unicode is my main terminal emulator because I think it's superior.
I am using transparent gaps because it looks just better for 99% of the stuff I'm going.. BUT
Sadly, nvim doesn't look good with border gaps.
It just doesn't look right with buffer & lualine, if you want a "complete" look.
With transparency applied u also have to either live with complete transparent *-lines or have opaque lines which destroys the aesthetics IMHO.
So long story short, I want to use st as a secondary terminal emulator solely to serve for nvim with 0 borders/gaps and I would like rxvt-unicode to swallow st with nvim whenever I call it within rxvt.
I have the swallow patch applied but rxvt doesn't swallow st.
And now that I'm thinking of it, I can find many more use cases for it, if I'm not limited to terminal swallowing. =)

**Comment by bakkeby** [comment]
I have seen that patch, it is quite massive. Never seen the point of manual swallowing. 

As for your use case you can achieve this two ways.

1) you could use the [devour](https://github.com/salman-abedin/devour) script

or 

2) use the normal swallow patch

The only reason why your xterm would not be swallowing st is because you have a client rule that says that st is a terminal. If you say that it is not a terminal then it should swallow fine.

If you want to keep st as a terminal that can swallow other windows, then just pass in a different class via command line arguments when starting st (I think it was the -e flag). That way you can have different rules for st depending on what class you pass in (or not).

**Comment by der-zonk** [comment]
So is that a no?

**Comment by bakkeby** [comment]
What I am saying is that your use case is already covered by what is there now. 

Have you actually used / tested that patch and found that this works really well for you or are you merely asking me to spend a lot of effort incorporating this into dwm-flexipatch just so that you can try it out?

I am not saying that it is a definite no-go to include this type of functionality, but I would most likely do it differently rather than taking that patch as-is. Not sure when I'll find time to look into this though, have quite a lot going on.

**Comment by der-zonk** [comment]
> What I am saying is that your use case is already covered by what is there now.
Can you swallow stuff with a terminal that has the "isfloating" rule applied?
I can't. =) Whatever is called goes right into tiled mode, leaving the floating terminal 100% unaffected.

**Comment by bakkeby** [comment]
Github doesn't send email notifications for comment edits so I missed this yesterday.

> And btw rxvt spits out errors when I give st another class and open and close nvim 

I presume you mean errors on this form?

```java
$ st
erresc: unknown csi ESC[22;0t
erresc: unknown csi ESC[22;0t
erresc: unknown csi ESC[>4;2m
erresc: unknown csi ESC[>4;0m
erresc: unknown csi ESC[23;0t
erresc: unknown csi ESC[23;0t
```

That is common, it has to do with unhandled escape code that nvim uses. If those are annoying then just redirect standard err to /dev/null.

```sh
$ st 2>/dev/null
```

> and it doesn't work with my floating rxvt at all so simply using an other class is not a satisfactory solution for me.

If swallowing works at all then it should not matter whether the terminal window is floating or not. `xclock` is a nice program to test with I find.

Installing rxvt-unicode, making urxvt a terminal and running a command like this works fine for me in my build.

```sh
$ st -c asdfasdf -e nvim 2>/dev/null
```

**Comment by der-zonk** [comment]
Yea I figured it out myself, hence I closed the issue.
